jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/KafkaWriter.java

```diff
@@ -0,0 +1,133 @@
+ * Kafka Topics can be passed as separated by commas such as kafka01,kafka02.
+ *
+ * @author utkarsh bhatnagar
+ */
+
+@NotThreadSafe
+public class KafkaWriter extends BaseOutputWriter {
+	
+	private static final Logger log = LoggerFactory.getLogger(KafkaWriter.class);
+	
+	private static final String DEFAULT_ROOT_PREFIX = "servers";
+	
+	private Producer<String,String> producer;
+	private String[] topics;
+	private boolean isTest;
+	private final String rootPrefix;
+
+	@JsonCreator
+	public KafkaWriter(
+			@JsonProperty("typeNames") ImmutableList<String> typeNames,
+			@JsonProperty("booleanAsNumber") boolean booleanAsNumber,
+			@JsonProperty("rootPrefix") String rootPrefix,
+			@JsonProperty("debug") Boolean debugEnabled,
+			@JsonProperty("settings") Map<String, Object> settings) {
+		super(typeNames, booleanAsNumber, debugEnabled, settings);
+		this.rootPrefix = resolveProps(
+				firstNonNull(
+						rootPrefix,
+						(String) getSettings().get("rootPrefix"),
+						DEFAULT_ROOT_PREFIX));		
+		// Setting all the required Kafka Properties
+		Properties kafkaProperties =  new Properties();
+		kafkaProperties.setProperty("metadata.broker.list", Settings.getStringSetting(settings, "metadata.broker.list", null));
+		kafkaProperties.setProperty("zk.connect", Settings.getStringSetting(settings, "zk.connect", null));
+		kafkaProperties.setProperty("serializer.class", Settings.getStringSetting(settings, "serializer.class", null));
+		this.producer= new Producer<String,String>(new ProducerConfig(kafkaProperties));
+		this.topics = Settings.getStringSetting(settings, "topics", null).split(","); 
+	}
+	
+	public void validateSetup(Server server, Query query) throws ValidationException {
+	}
+
+	public void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
+
+		try {
+			ArrayList<KeyedMessage<String,String>> messages = new ArrayList<KeyedMessage<String,String>>();
+			List<String> typeNames = this.getTypeNames();
+
+			for (Result result : results) {
+				log.debug("Query result: {}", result);
+				Map<String, Object> resultValues = result.getValues();
+				if (resultValues != null) {
+					for (Entry<String, Object> values : resultValues.entrySet()) {
+						Object value = values.getValue();
+						if (NumberUtils.isNumeric(value)) {
+							JSONObject obj = new JSONObject();
+							obj.put("keyspace",KeyUtils.getKeyString(server, query, result, values, typeNames, this.rootPrefix).replaceAll("[()]", "_"));
+							obj.put("value", value.toString());
+							obj.put("timestamp", result.getEpoch() / 1000);
+							for(String topic : this.topics) {
+								log.debug("Topic: "+topic+" ; Kafka Message: {}", obj.toString());
+								messages.add(new KeyedMessage<String,String>(topic,obj.toString()));
+							}
+						} else {
+							log.warn("Unable to submit non-numeric value to Kafka: [{}] from result [{}]", value, result);
+						}
+					}
+				}
+			}
+			if (!this.isTest)
```

Comment:
This `isTest` condition is dubious. If you want a way to unit test this class, you should provide a way to inject the `producer` and mock it.